src/element-transformer/index.ts

```diff
@@ -0,0 +1,132 @@
+import * as ts from 'typescript';
+
+export default function transformer(program: ts.Program): ts.TransformerFactory<ts.SourceFile> {
+	return (context: ts.TransformationContext) => (file: ts.SourceFile) => visitNodeAndChildren(file, program, context);
+}
+
+function visitNodeAndChildren(f: ts.SourceFile, program: ts.Program, context: ts.TransformationContext): ts.SourceFile;
+function visitNodeAndChildren(node: ts.Node, program: ts.Program, context: ts.TransformationContext): ts.Node;
+function visitNodeAndChildren(node: ts.Node, program: ts.Program, context: ts.TransformationContext) {
+	return ts.visitEachChild(
+		decoratorVisit(node, program),
+		(childNode) => visitNodeAndChildren(childNode, program, context),
+		context
+	);
+}
+
+function decoratorVisit(node: ts.Node, program: ts.Program): ts.Node {
+	if (!(ts.isClassDeclaration(node) && node.decorators && node.name)) {
+		return node;
+	}
+	const decorator = node.decorators.find((dec) => {
+		return ts.isCallExpression(dec.expression) && dec.expression.expression.getText() === 'customElement';
```

Comment:
is this enough? what happens if they aliased the import name for the customElement decorator?

Reply:
you won't need this anyway if you no longer need to find the decorator in the first place for munging the objects

Reply:
Then this transform will fail. I did look through a bunch of transformers and pretty much all of them work directly off the name for expedience.

It should be possible to get the exported symbol for the decorator and resolve it with the symbol for this identifier but I really don't have a good feel for how type checking and symbol resolution work in the compiler. I'd also have to redo the unit tests to get the decorator to resolve. I expect I'll need to take a couple evenings digging through the compiler source to figure things out.

Reply:
as i said, if you just don't do the object mangling you never need to find an existing decorator anyway